common/djangoapps/third_party_auth/signals/handlers.py

```diff
@@ -0,0 +1,55 @@
+"""
+Signal handlers for third_party_auth app.
+"""
+
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from ..models import SAMLConfiguration, SAMLProviderConfig
+
+log = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=SAMLConfiguration)
+def update_saml_provider_configs_on_configuration_change(sender, instance, created, **kwargs):
+    """
+    Signal handler to update SAMLProviderConfig instances when SAMLConfiguration is updated.
+    
+    When a SAMLConfiguration is updated, ConfigurationModel creates a new version.
+    This handler ensures that all SAMLProviderConfig instances that were using
+    the old configuration are updated to point to the new version.
+    """
+    if not instance.enabled:
+        return
```

Comment:
I was thinking this needed a comment, but maybe this needs to go away? If we point everything to the disabled version, we just need to ensure that it will provide the fallback of the default if this is disabled. And then if it is later enabled, it will update all the same providers to continue pointing to the same slug and to now be enabled.

Reply:
I think you showed that the Admin UI will display a disabled option, so this should be all set. Marking resolved.